xrmocap/utils/data_convert_utils.py

```diff
@@ -0,0 +1,248 @@
+# yapf: disable
+import logging
+import numpy as np
+from enum import Enum
+from mmhuman3d.data.data_structures.human_data import HumanData
+from typing import Optional, Union
+
+from xrmocap.data_structure.body_model import SMPLData, SMPLXData
+
+# yapf: enable
+
+
+class SMPLDataTypeEnum(str, Enum):
+    SMPLDATA = 'smpl data'
+    HUMANDATA = 'human data'
+    AMASS = 'AMASS'
+    UNKNOWN = 'unknown'
+
+
+def validate_shape(actual_shape: tuple, expected_shape: tuple) -> bool:
+    """Compare the shape of two ndarrays.
+
+    Args:
+        actual_shape (tuple): the actual shape.
+        expected_shape (tuple): the expected shape.
+
+    Returns:
+        bool: returns true if the actual shape is the expected shape.
+    """
+    return all(a == e or e is None
+               for a, e in zip(actual_shape, expected_shape))
+
+
+def validate_spec(specs: dict, data: dict) -> bool:
+    """Validate whether the input data conform to the specs.
+
+    Args:
+        specs (dict): rules that should be followed.
+        data (dict): data to be evaluated.
+
+    Returns:
+        bool: returns true if the data follows the specs.
+    """
+    missing_keys = set(specs.keys()) - set(data.keys())
+    if missing_keys:
+        return False
+
+    for key, expected_shape in specs.items():
+        item = data[key]
+        if not validate_shape(item.shape, expected_shape):
+            return False
+    return True
+
+
+class SMPLDataConverter:
```

Comment:
We've already got [data converters](https://github.com/openxrlab/xrmocap/blob/main/xrmocap/data/data_converter/panoptic_data_converter.py) for dataset, shall we have a different name here?